src/irmin/tree.ml

```diff
@@ -1888,7 +2047,7 @@ module Make (P : Private.S) = struct
       in
       if List.compare_length_with elts len = 0 then
         (* We have a complete proof, let's build a tree node directly! *)
-        k (Node.of_map (StepMap.of_seq (List.to_seq elts)))
+        k (Node.of_map ~env (StepMap.of_seq (List.to_seq elts)))
```

Comment:
Could we drop the `complete` branch? The `else` branch could handle the `complete` case too